ratelimit/doc.go

```diff
@@ -0,0 +1,93 @@
+/*
+Package ratelimit implements rate limiting functionality for the proxy.
+
+It provides per instance based, local, rate limiting. It can be
+configured globally, or based on routes. Rate limiting can be lookuped
+based on HTTP headers like X-Forwarded-For or Authorization.
+
+Lookuper Type - Authorization Header
```

Comment:
This seems to be bucketing based on Auth header, is that it?

Reply:
bucketing? This sounds more weird than lookup to me. 
You also "lookup" a value from a hashmap, not "bucket" or "bucketing" a value from a hashmap.